lib/test/t_LeastSquaresExpansion_std.cxx

```diff
@@ -0,0 +1,158 @@
+                Point ref(coeffs.getSize());
+                std::copy(expectedCoefficientsLinear.begin(), expectedCoefficientsLinear.begin() + coeffs.getSize(), ref.begin());
+                Scalar err = (coeffs - ref).norm();
+                const Scalar rtol = 1.0e-3;
+                const Scalar atol = 1.0e-3;
+                assert_almost_equal(err, 0.0, rtol, atol);
+              }
+              if (wMin == wMax)
+                {
+                  // Create the polynomial chaos algorithm using the full constructor
+                  LeastSquaresExpansion algo(inputSample, outputSample, distribution, productBasis, basisSize, methodNames[n]);
+                  algo.run();
+                  // Check the coefficients
+                  FunctionalChaosResult result(algo.getResult());
+                  Point coeffs(result.getCoefficients().asPoint());
+                  Point ref(coeffs.getSize());
+                  std::copy(expectedCoefficientsLinear.begin(), expectedCoefficientsLinear.begin() + coeffs.getSize(), ref.begin());
+                  Scalar err = (coeffs - ref).norm();
+                  const Scalar rtol = 1.0e-3;
+                  const Scalar atol = 1.0e-3;
+                  assert_almost_equal(err, 0.0, rtol, atol);
+                }
+            } // Method
+          {
+            // Create the polynomial chaos algorithm using the simplified constructor
+            LeastSquaresExpansion algo(inputSample, weights, outputSample, distribution);
+            algo.run();
+            // Check the coefficients
+            FunctionalChaosResult result(algo.getResult());
+            Point coeffs(result.getCoefficients().asPoint());
+            Point ref(coeffs.getSize());
+            std::copy(expectedCoefficientsHyper.begin(), expectedCoefficientsHyper.begin() + coeffs.getSize(), ref.begin());
+            Scalar err = (coeffs - ref).norm();
+            const Scalar rtol = 1.0e-2;
+            const Scalar atol = 1.0e-2;
+            assert_almost_equal(err, 0.0, rtol, atol);
+          }
+          // Check the constructors assuming uniform weights
+          if (wMin == wMax)
+            {
+              // Create the polynomial chaos algorithm using the simplified constructor
+              LeastSquaresExpansion algo(inputSample, outputSample, distribution);
+              algo.run();
+              // Check the coefficients
+              FunctionalChaosResult result(algo.getResult());
+              Point coeffs(result.getCoefficients().asPoint());
+              Point ref(coeffs.getSize());
+              std::copy(expectedCoefficientsHyper.begin(), expectedCoefficientsHyper.begin() + coeffs.getSize(), ref.begin());
+              Scalar err = (coeffs - ref).norm();
+              const Scalar rtol = 1.0e-2;
+              const Scalar atol = 1.0e-2;
+              assert_almost_equal(err, 0.0, rtol, atol);
+            } // Uniform weights
+        } // i
```

Comment:
There is no unit test of the `setActiveFunctions` here. It should be added, so that we can see that the method actually performs correctly.

Reply:
The test of the `setActiveFunctions` should be done, at least _a minima_, to see that the evaluation does not fail.